tests/k3d/dev-cluster.sh

```diff
@@ -0,0 +1,184 @@
+#!/usr/bin/env bash
+
+# dev-cluster.sh manages k3d clusters for development and testing.
+# It supports creating and deleting clusters with configurable nodes, versions, and regions.
+
+set -euo pipefail
+
+# Default configuration
+DEFAULT_REGION="us-east-1"
+DEFAULT_ZONES=3
+DEFAULT_NODES=1
+DEFAULT_K8S_VERSION="1.30.6"
+DEFAULT_CLUSTER_NAME="local"
+
+# Binary paths
+K3D_PATH="./bin/k3d"
+
+# Registry configuration
+REGISTRY="gcr.io"
+REPOSITORY="cockroachlabs-helm-charts/cockroach-self-signer-cert"
+
+# Required container images for the cluster.
+# These images are imported into the cluster during creation.
+REQUIRED_IMAGES=(
+    "quay.io/jetstack/cert-manager-cainjector:v1.11.0"
+    "quay.io/jetstack/cert-manager-webhook:v1.11.0"
+    "quay.io/jetstack/cert-manager-controller:v1.11.0"
+    "quay.io/jetstack/cert-manager-ctl:v1.11.0"
+    "cockroachdb/cockroach:v24.3.5"
+    "${REGISTRY}/${REPOSITORY}:$(bin/yq '.tls.selfSigner.image.tag' ./cockroachdb/values.yaml)"
+)
```

Comment:
We are not using the local registry anywhere. We are building the selfSigner image in the CI process itself and not pushing it to local registry. This image as we are building in same sandbox it is present there as it is.

As image is present in the sandbox environment so in func `import_container_images` the docker inspect command doesn't return non zero return code and hence docker pull command doesn't run.

This entire workflow will work without the local registries as well